app/src/Controller/ApiController.php

```diff
@@ -0,0 +1,146 @@
+<?php
+
+namespace App\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpFoundation\JsonResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use App\Api\GameApi;
+use App\Api\TokenGenerator;
+use App\Entity\Game;
+use App\Entity\Player;
+use App\Repository\GameRepository;
+use App\Repository\PlayerRepository;
+use App\ArgumentResolver\GameContext;
+
+class ApiController extends Controller {
+  public const DEFAULT_SIZE = 4;
+
+  private $serializer;
+  private $api;
+  private $gameRepository;
+  private $playerRepository;
+
+  public function __construct(GameApi $gameApi, GameRepository $gameRepository, PlayerRepository $playerRepository) {
+    $this->api = $gameApi;
+    $this->gameRepository = $gameRepository;
+    $this->playerRepository = $playerRepository;
+  }
+
+  public function new(Request $request) {
+    $mode = $request->get('mode');
+    $apiResponse = $this->api->new(self::DEFAULT_SIZE, $mode && $mode == 'multi');
+    $this->playerRepository->save($apiResponse['player']);
```

Comment:
As already said at presentation, you use our repositories in a "active record" way.
You don't benefit from Doctrine ORM DataMapper that let you save all our "persisted" entities in the same time.

You do:
->persist(player)
->flush() => Save to database
->persist(game)
->flush() => Save to database

You must preferably do:
->persist(player)
->persist(game)
->flush() => Save to database in only one transaction